### PR TITLE
Adding vulnerability Assessemtn clinet code

### PR DESCRIPTION
**Pull request by yaakoviyun**
Adding vulnerability Assessemtn clinet code

### CODE REVIEW

**Comment by yaakoviyun** [comment]
@jaredmoo - please review too...

**Review by jaredmoo — CHANGES_REQUESTED** [review]
Overall looking pretty good :)

**Review by dsgouda — CHANGES_REQUESTED** [review]
These look like additive changes. Please bump the minor version in csproj file along with PackageReleaseNotes

File: src/SDKs/SqlManagement/Sql.Tests/DatabaseVulnerabilityAssessmentScenarioTests.cs [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+                            },
+                            new DatabaseVulnerabilityAssessmentRuleBaselineItem()
+                            {
+                                Result = new string[] { "userB", "SELECT" }
+                            }
+                        });
+
+                sqlClient.DatabaseVulnerabilityAssessmentRuleBaselines.CreateOrUpdate(resourceGroup.Name, server.Name, dbName, testRuleId, ruleBaselineToSet);
+
+                // Get basleine
```
[the remark this answers is not in the document]
Comment (yaakoviyun) [reply src/SDKs/SqlManagement/Sql.Tests/DatabaseVulnerabilityAssessmentScenarioTests.cs]: Thanks, fixed.

File: src/SDKs/SqlManagement/Sql.Tests/DatabaseVulnerabilityAssessmentScenarioTests.cs [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+                await SetPolicy(context, sqlClient, resourceGroup, server, dbName);
+
+                // Run some scans
+                string scanId = string.Format("scantest1_{0}", DateTime.UtcNow.Ticks);
+                sqlClient.DatabaseVulnerabilityAssessmentScans.Execute(resourceGroup.Name, server.Name, dbName, scanId);
+
+                string scanId1 = string.Format("scantest2_{0}", DateTime.UtcNow.Ticks);
+                sqlClient.DatabaseVulnerabilityAssessmentScans.Execute(resourceGroup.Name, server.Name, dbName, scanId1);
+
+                // Veroify get scan and list scans
```
[the remark this answers is not in the document]
Comment (yaakoviyun) [reply src/SDKs/SqlManagement/Sql.Tests/DatabaseVulnerabilityAssessmentScenarioTests.cs]: Thanks, fixed.

File: src/SDKs/SqlManagement/Sql.Tests/DatabaseVulnerabilityAssessmentScenarioTests.cs [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+                {
+                    Location = server.Location,
+                });
+                Assert.NotNull(db1);
+
+                // Set policy
+                await SetPolicy(context, sqlClient, resourceGroup, server, dbName);
+
+                // Run some scans
+                string scanId = string.Format("scantest1_{0}", DateTime.UtcNow.Ticks);
```
[the remark this answers is not in the document]
Comment (yaakoviyun) [reply src/SDKs/SqlManagement/Sql.Tests/DatabaseVulnerabilityAssessmentScenarioTests.cs]: Thanks, fixed.

File: src/SDKs/SqlManagement/Sql.Tests/DatabaseVulnerabilityAssessmentScenarioTests.cs [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+                sqlClient.DatabaseVulnerabilityAssessmentScans.Execute(resourceGroup.Name, server.Name, dbName, scanId);
```
[the remark this answers is not in the document]
Comment (yaakoviyun) [reply src/SDKs/SqlManagement/Sql.Tests/DatabaseVulnerabilityAssessmentScenarioTests.cs]: yep, it was.
will change the client code and update the swagger in azure-rest-api-specs repo

File: src/SDKs/SqlManagement/Management.Sql/Generated/Models/DatabaseVulnerabilityAssessment.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        [JsonProperty(PropertyName = "properties.storageContainerPath")]
+        public string StorageContainerPath { get; set; }
+
+        /// <summary>
+        /// Gets or sets a shared access signature (SAS Key) that has write
+        /// access to the blob container specified in 'storageContainerPath'
+        /// parameter.
+        /// </summary>
+        [JsonProperty(PropertyName = "properties.storageContainerSasKey")]
+        public string StorageContainerSasKey { get; set; }
```
[the remark this answers is not in the document]
Comment (yaakoviyun) [reply src/SDKs/SqlManagement/Management.Sql/Generated/Models/DatabaseVulnerabilityAssessment.cs]: Yep, will fix

**Comment by jaredmoo** [comment]
@dsgouda - Jenkins build failure below seems like not our fault

```
00:35:56   c:\workspace\NetCore-SdkCI\tools\SdkBuildTools\targets\common.targets(215,5): error MSB3073: The command "dotnet test -f netcoreapp1.1 c:\workspace\NetCore-SdkCI\src\SdkCommon\TestFramework\HttpRecorder.Tests\HttpRecorder.Tests.csproj -l trx;LogFileName=c:\workspace\NetCore-SdkCI\TestResults\netCore11\HttpRecorder.Tests.trx " exited with code 1. [c:\workspace\NetCore-SdkCI\build.proj]
```

**Review by jaredmoo — APPROVED** [review]

**Comment by yaakoviyun** [comment]
@dsgouda , can you merge?

**Comment by jaredmoo** [comment]
This PR was merged into #4351 and can be closed,

**Pull request merged**